CodeGeneration/Sources/generate-swiftsyntax/templates/basicformat/BasicFormatFile.swift

```diff
@@ -122,6 +122,20 @@ let basicFormatFile = SourceFile {
     }
 
     FunctionDecl("open func requiresLeadingSpace(_ token: TokenSyntax) -> Bool") {
+      SwitchStmt("""
+        switch (token.previousToken(viewMode: .sourceAccurate)?.tokenKind, token.tokenKind) {
+        case (.identifier(_), .leftAngle),
+             (.identifier(_), .rightAngle),
+             (.postfixQuestionMark, .leftAngle),
+             (.postfixQuestionMark, .rightAngle):
```

Comment:
Don’t we need the same for 
- exclamation marks for implicitly unwrapped optional types (`MyGeneric<Foo?>`)
- right square for array and dictionary types (`MyGeneric<[Foo]>`)

Incidentally, what breaks if we unconditionally don’t require leading space for `<` and `>`?

Reply:
If we remove trailing and leading space from `<` and `>` the test suite pass

Reply:
Do you mean the test suite doesn't pass? What exactly break?

Reply:
Nothing, all the tests are green ✅

Reply:
In that case, I would prefer to just change to set `requires_leading_space` to `False` for `<` and `>`.

Reply:
I see that you already changed that. In that case, I think you can remove the special cases here.

Reply:
The code was outdated. It should be alright now.

Reply:
@ahoppen will you resolve this if you agree?